app/src/main/assets/shaders/simulation/advection/advection.frag

```diff
@@ -6,23 +6,24 @@ precision highp sampler3D;
 layout(binding = 0) uniform sampler3D velocity_field;
 layout(binding = 1) uniform sampler3D data_field;
 
-uniform float dt;
-uniform int depth;
-uniform vec3 gridSize;
+uniform float dt;   //in seconds
+uniform float meterToPixels;  //conversion factor from meter to pixels
```

Comment:
should be grids not pixels

Reply:
or texels or voxels, but not pixels since it is not written to screen. A bit misleading only